### DISCUSSION TITLE
Multiple Monitors (different sizes) - nixOS host - microsoft client on MacOS

### DISCUSSION THREAD

**Discussion by gotjoshua** [neutrinolabs/xrdp#3036] [Q&A]
After quite some trial and error, and awkwardness, I managed to get remote gnome desktop working on my NixOS host.

But it only works with a single monitor, and only if one user is logged in on the actual local machine, but not the user who will login remotely.

I will collect some info here about the multiple monitor situation, in hopes that someone can help me solve it!

**Comment by gotjoshua** [comment]
In the user's home directory who is logging in (kori):
via:
` sudo cat /home/kori/.xorgxrdp.1.log`
```
[  5539.266] rdpRRGetInfo:
[  5553.937] rdpClientConProcessMsgClientInfo:
[  5553.937]   got client info bytes 7840
[  5553.937]   jpeg support 0
[  5553.937]   offscreen support 0
[  5553.937]   offscreen size 0
[  5553.937]   offscreen entries 0
[  5553.937]   client can not do offscreen to offscreen blits
[  5553.937]   client can do new(color) cursor
[  5553.937]   client can do multimon
[  5553.938]   client monitor data, monitorCount=2
[  5553.938]     left 0 top 1080 right 1279 bottom 1879
[  5553.938]     left 0 top 0 right 1919 bottom 1079
[  5553.938] rdpRRSetRdpOutputs: numCrtcs 0 numOutputs 0 monitorCount 2
[  5553.938] rdpRRSetRdpOutputs: add output 0 left 0 top 1080 width 1280 height 800
[  5553.938] rdpRRSetRdpOutputs: add output 1 left 0 top 0 width 1920 height 1080
[  5553.938] rdpLoadLayout: keylayout 0x00000000 variant  display 1
```

then in the syslogs via
`journalctl -f`
```
 Starting GNOME Shell on X11...
[42149]: Running GNOME Shell (using mutter 45.5) as a X11 window and compositing manager
64]: rdpRRGetInfo:
49]: DRI3 not available
49]: failed to load driver: zink
64]: rdpkeybChangeKeyboardControl:
64]: rdpkeybChangeKeyboardControl: autoRepeat on
64]: rdpkeybChangeKeyboardControl:
64]: rdpkeybChangeKeyboardControl: autoRepeat on
49]: Xlib:  extension "DPMS" missing on display ":1.0".
64]: rdpRRCrtcGetGamma: 0x2b12d50 0x2b13040 0x2b13440 0x2b13240
64]: rdpRRCrtcSet:
64]: rdpRRCrtcSet:
49]: Window manager warning: Configuring CRTC 1024 with mode 1023 (1920 x 1080 @ 50.000000) at position 1280, 0 and transform 0 failed
[42149]: Received an X Window System error.
         This probably reflects a bug in the program.
         The error was 'BadMatch (invalid parameter attributes)'.
           (Details: serial 273 error_code 8 request_code 140 (RANDR) minor_code 7)
           (Note to programmers: normally, X errors are reported asynchronously;
            that is, you will receive the error a while after causing it.
            To debug your program, run it with the MUTTER_SYNC environment
            variable to change this behavior. You can then get a meaningful
            backtrace from your debugger if you break on the mtk_x_error() function.)
```

**Comment by matt335672** [comment]
@gotjoshua - [this comment](https://github.com/neutrinolabs/xrdp/issues/2579#issuecomment-1475936282) might help. Check your software versions.

**Reply by gotjoshua** [reply]
/nix/store/01zd6sxkym5a3aj6miwn2ip1id57bn82-xrdp-0.9.25.1/bin/xrdp
 /nix/store/01zd6sxkym5a3aj6miwn2ip1id57bn82-xrdp-0.9.25.1/bin/xrdp-sesman


  xorgxrdp = stdenv.mkDerivation rec {
    pname = "xorgxrdp";
[    version = "0.9.20";](https://github.com/NixOS/nixpkgs/blob/8e54038847aa80ee2b1ffa9e313b0ab918f1903c/pkgs/applications/networking/remote/xrdp/default.nix#L40)

**Reply by gotjoshua** [reply]
[back in 2022](https://github.com/neutrinolabs/xrdp/issues/1969#issuecomment-1048961196) the following versions were recommended... but what about now? should i downgrade by 2 years?

> ./xorgxrdp_0.2.15-1_amd64.deb 
> ./xrdp_0.9.15-1_amd64.deb

**Comment by matt335672** [comment]
Your display config looks interesting. From the log I've got `:1.0`.

A normal xrdp config won't use a display as low as 1. Have you changed the config to set this? If not, this could be related to a second login on your primary screen.

Also, the screen 0 suffix can probably be removed. I don't know how our code would react to that.

**Reply by gotjoshua** [reply]
good catch. i removed a patch that replaced the `X11DisplayOffset=10` with   `X11DisplayOffset=0` but that didn't solve it (yet)

Now i am in the middle of trying to update the nixOS pkg to use 0.10   (results may vary)
